content/blog/2022-05-03-e2e-cv-pt1.md

```diff
@@ -0,0 +1,307 @@
+---
+title:
+  'End-to-End Computer Vision API, Part 1: Data Versioning and ML pipelines'
+date: 2022-05-03
+description: >
+
+  In the last few years, training a well-performing Computer Vision (CV) model
+  in Jupyter Notebooks became fairly straightforward if you use pre-trained Deep
+  Learning models and high-level libraries that abstract away much of the
+  complexity (`fastai`, `keras`, `pytorch-lightning` are just a few
+  examples).  The hard part is still incorporating this model into an
+  application that runs in a production environment bringing value to the
+  customers and our business.
+
+descriptionLong: |
+  _This is the first part of a three-part series of posts._
+
+  In the last few years, training a well-performing Computer Vision (CV) model in Jupyter Notebooks became fairly straightforward if you use pre-trained Deep Learning models and high-level libraries that abstract away much of the complexity (`fastai`, `keras`, `pytorch-lightning` are just a few examples). 
+  The hard part is still incorporating this model into an application that runs in a production environment bringing value to the customers and our business.
+picture: 2022-05-03/e2e-cv-pt1-cover.png
+author: alex_kim
+commentsUrl: https://discuss.dvc.org/t/end-to-end-computer-vision/1178
+tags:
+  - Computer Vision
+  - DVC
+  - CML
+  - Studio
+  - CI/CD
+  - Experiment Tracking
+---
+
+### Introduction
+
+A typical ML project lifecycle goes through three phases:
+
+1. Active exploration or proof-of-concept phase. Here we try many different
+   approaches to data preprocessing, model architectures, and hyperparameter
+   tuning with the goal of finally settling on the most promising combination.
+2. Application development phase. Here we build all the “plumbing” around the
+   model, that is: getting input data, massaging it into the right format,
+   passing it to the model, and, finally, serving the model's output. The goal
+   here is to end up with a version of an end-to-end application that is robust
+   and well-performing enough to be considered for deployment to the production
+   environment.
+3. Production deployment phase. Here we promote a well-tested version of the
+   application from the development environment to the production environment.
+
+In this series of posts we'll describe an approach that streamlines all three
+phases. As our demo project, we've selected a very common deployment pattern in
+CV projects: a CV model wrapped in a web API service. Automatic defect detection
+is an example problem that we are addressing with this pattern. Specifically,
+we'll perform an
+[image segmentation](https://ai.stanford.edu/~syyeung/cvweb/tutorial3.html) task
```

Comment:
The deployment pattern addresses automatic defect detection?
Image segmentation is a kind of automatic defect detection?
Sorry, I have no experience in this field. But maybe we can water down the introduction for a more general reader? Not sure

Reply:
Added a paragraph with more context